include/appimage/update.h

```diff
@@ -1,124 +1,133 @@
+        // Beware that it will return true in case of errors, too! Combine with either state() or hasError()!
+        bool isDone();
+
+        // Convenience function returning whether an error has occured. Uses state() internally.
+        // Beware that it will return false even if the update process has not yet begun, or is currently running!
+        bool hasError();
+
+        // Sets given parameter to current progress. Returns false in case of failure, i.e., the update process
+        // is not running or the version of the AppImage format is not supported, otherwise true.
+        bool progress(double& progress);
+
+        // Fetch a status message from the client in use that can be used to display updates
+        bool nextStatusMessage(std::string& message);
+
+        // Check whether an update is available
+        // Please note that this method is *only* available until the update is started (after calling start(),
+        // the method will instantly return false)
+        bool checkForChanges(bool& updateAvailable, unsigned int method = 0);
+
+        // Parses AppImage file, and returns a formatted string describing it
+        // in case of success, sets description and returns true, false otherwise
+        bool describeAppImage(std::string& description) const;
+
+        // Sets path to the path of the file created by the update and returns true as soon as this value is
+        // available (after a successful update at the latest)
+        // Returns false in case of errors, or when the path is not available yet
+        bool pathToNewFile(std::string& path) const;
+
+        // Validate AppImage signature
+        // TODO: describe process
+        // Returns a ValidationState value. See ValidationState documentation for more information.
+        ValidationState validateSignature();
+
+        // Returns a description string of the given validation state.
+        static std::string signatureValidationMessage(const ValidationState& state);
+
+        // Returns the size of the remote file in bytes
+        bool remoteFileSize(off_t& fileSize) const;
+
+        // Return update information
+        // If no custom update information is set, the function returns the raw update information contained in the
+        // AppImage
+        [[nodiscard]] std::string updateInformation() const;
+
+        // Set custom update information
+        // The custom update information will overwrite the one fetched from the AppImage
+        // This function is for applications that use libappimageupdate, e.g., to self-update, or implement a custom
+        // frontend like AppImageLauncher does, extending those users' possibilities
+        // Use cases include the use of custom update servers instead of relying on a built-in update information type,
+        // implement multiple update channels, etc.
+        void setUpdateInformation(std::string newUpdateInformation);
```

Comment:
Indeed, this is a good idea, since the class has async features. I'll have to synchronize it, as well as make sure the update information can only be changed until the process has started.